config/initializers/docker.rb

```diff
@@ -1,7 +1,7 @@
 # Docker specific development configuration
 if Rails.env.development? && File.file?("/.dockerenv")
   # Using shell tools so we don't need to require Socket and IPAddr
-  host_ip = `/sbin/ip route|awk '/default/ { print $3 }'`.strip
+  host_ip = `ip route|awk '/default/ { print $3 }'`.strip
```

Comment:
This won't work on MacOS: I think `ifconfig` is the equivalent: https://unix.stackexchange.com/questions/319350/ip-command-of-iproute2-package

Reply:
@joshpuetz This doesn't need to work on  macOS, as it will only run when the app starts in a Docker container (that's the `File.file?("/.dockerenv")` part).

Reply:
Ah, good point! That makes the error `sh: ip: command not found` all that more confounding!

Reply:
@joshpuetz make sure you rebuilt the containers as the Containerfile changed with the iproute package being added. I think you are still using the older Container that doesn't have the `iproute` package installed. 

This whole part most likely should be refactored so it just uses Ruby to figure out the default route vs shelling out to `ip` or just removed. I am not really sure what we are whitelisting here but hard coding paths to system binaries in the app makes the code less portable.

Reply:
@jdoss We are adding these IPs to the list of allowed addresses for Better Errors and the web console, otherwise the debugging experience in Docker will be worse than locally. 

IMHO the hard-coded path was not a problem, as this only applies to the one environment we actually fully control, the distribution running inside the container. The only time it stops working is when we switch that, as right now. I have no problem with doing it in Ruby either, the comment already hints at the two libraries (`Socket` and `IPAddr`) you'll need for that.

Reply:
@jdoss As a baseline to get you started:

```ruby
Socket.ip_address_list.detect{ |intf| intf.ipv4_private? }
```

This will return `Addrinfo` objects, calling `ip_address` on those will return the IPs as strings.

Reply:
My latest commit should fix the issue with `ip` not being found. A total PEBKAC on my end.

```
[jdoss@sts7 devto]$ podman exec -it devto_web_1 /bin/bash
[root@3d439f7595ff devto]# ip route|awk '/default/ { print $3 }'
10.0.2.2
```